### PR TITLE
support mongodb on k8s

### PR DESCRIPTION
**Pull request by jsirianni**
- Added source parameter for selecting file or k8s input
- Added cluster, container, pod inputs when using kubernetes
- Set non json format output to regex, which was implicitly set before due to pipeline order

```json
{
  "level": "info",
  "timestamp": "2021-07-21T17:59:09.217Z",
  "message": "Starting stanza agent"
}

{
  "level": "info",
  "timestamp": "2021-07-21T17:59:09.428Z",
  "message": "Stanza agent started"
}

{
  "level": "info",
  "timestamp": "2021-07-21T17:59:09.629Z",
  "message": "Started watching file from end. To read preexisting logs, configure the argument 'start_at' to 'beginning'",
  "operator_id": "$.mongodb.kubernetes_input.container_reader",
  "operator_type": "file_input",
  "path": "/var/log/containers/mongodb-5cc4698795-zvfbg_default_mongodb-b346c5f0b7c004aca6e56cdbea0ab1d4dc27075031cfb554a8501f70accd655b.log"
}

{
  "timestamp": "2021-07-21T17:59:14.894Z",
  "severity": 30,
  "severity_text": "I",
  "labels": {
    "k8s-pod/app": "mean-stack",
    "k8s-pod/component": "mongodb",
    "k8s-pod/pod-template-hash": "5cc4698795",
    "log_type": "k8s.container",
    "plugin_id": "kubernetes_input",
    "stream": "stdout"
  },
  "resource": {
    "k8s.cluster.name": "",
    "k8s.container.id": "b346c5f0b7c004aca6e56cdbea0ab1d4dc27075031cfb554a8501f70accd655b",
    "k8s.container.name": "mongodb",
    "k8s.deployment.name": "mongodb",
    "k8s.namespace.name": "default",
    "k8s.namespace.uid": "08963f4e-7ef0-44a5-9ba6-ca18b14c7969",
    "k8s.node.name": "",
    "k8s.pod.name": "mongodb-5cc4698795-zvfbg",
    "k8s.pod.uid": "efb20850-8323-41da-82af-e8b67bc0dd40",
    "k8s.replicaset.name": "mongodb-5cc4698795"
  },
  "record": {
    "attr": {
      "message": "[1626890354:894490][1:0x7f708bf03700], WT_SESSION.checkpoint: [WT_VERB_CHECKPOINT_PROGRESS] saving checkpoint snapshot min: 165, snapshot max: 165 snapshot count: 0, oldest timestamp: (0, 0) , meta checkpoint timestamp: (0, 0)"
    },
    "component": "STORAGE",
    "context": "Checkpointer",
    "id": 22430,
    "message": "WiredTiger message"
  }
}
```

### CODE REVIEW

File: plugins/mongodb.yaml [hunk 1]
```diff
@@ -22,12 +62,34 @@ parameters:
...
+  # {{ if eq $source "kubernetes" }}
+  - id: kubernetes_input
+    type: kubernetes_container
+    pod_name: '{{ $pod_name }}'
+    container_name: '{{ $container_name }}'
+    cluster_name: '{{ $cluster_name }}'
+    start_at: '{{ $start_at }}'
+    enable_nested_json_parser: false
+    # {{ if $mongodb_json_log_format }}
+    write_to: log
```
Comment (ericwholt) [inline plugins/mongodb.yaml:84]:
```suggestion
```
Looks like your  json_parser parses directly from record.

**Review by ericwholt — CHANGES_REQUESTED** [review]
Looks good!

File: plugins/mongodb.yaml (hunk 1 above)
Comment (jsirianni) [reply plugins/mongodb.yaml]: Good catch. Initially I was going to add the write_to option to the container plugin, but I opted to remove $record.log from the mongo plugin instead: https://github.com/observIQ/stanza-plugins/pull/293

**Review by ericwholt — APPROVED** [review]
Looks good to me!

**Review by schmikei — APPROVED** [review]

**Pull request merged**